addons/website_sale/static/src/js/website_sale_delivery.js

```diff
@@ -203,6 +203,10 @@ publicWidget.registry.websiteSaleDelivery = publicWidget.Widget.extend({
         const address = selectedCarrierEl.querySelector('.o_order_location_address').innerText
         const isPickUp = selectedCarrierEl.lastChild.previousSibling.children;
```

Comment:
if the carrier has a description and pickup points, we still want to run the code I think (with the current solution we skip the code in this case)

for this solution, I think we could instead change the way we get isPickup so we always get the right one, eg. 

```suggestion
        const isPickUp = selectedCarrierEl.querySelector('.o_order_location').parentNode.parentNode.children;
```
and in master we could add a class to:

https://github.com/odoo/odoo/blob/70087feefb3ff629fa32826b823dce8910ebd700/addons/website_sale/views/website_sale_delivery_templates.xml#L26

and directly target that class